data/json/mapgen/bugs/spider_pit.json

```diff
@@ -5,35 +5,35 @@
     "om_terrain": [ "spider_pit" ],
     "weight": 250,
     "object": {
+      "predecessor_mapgen": "forest",
       "rows": [
-        "........................",
-        "........................",
-        "........................",
-        "........................",
-        "........................",
-        "........................",
-        "........................",
-        "........................",
-        "........................",
-        "........................",
-        "........................",
-        "........................",
-        "............>...........",
-        "........................",
-        "........................",
-        "........................",
-        "........................",
-        "........................",
-        "........................",
-        "........................",
-        "........................",
-        "........................",
-        "........................",
-        "........................"
+        "                        ",
+        "                        ",
+        "                        ",
+        "                        ",
+        "                        ",
+        "                        ",
+        "                        ",
+        "                        ",
+        "                        ",
+        "                        ",
+        "                        ",
+        "           ...          ",
+        "           .>.          ",
+        "           ...          ",
+        "                        ",
+        "                        ",
+        "                        ",
+        "                        ",
+        "                        ",
+        "                        ",
+        "                        ",
+        "                        ",
+        "                        ",
+        "                        "
       ],
-      "terrain": { ".": [ [ "t_region_groundcover_forest", 50 ], [ "t_region_shrub", 2 ], "t_region_tree" ], ">": "t_slope_down" },
-      "traps": { ".": [ [ "tr_null", 20 ], "tr_sinkhole" ] },
```

Comment:
Was removing the sinkholes intentional?

Reply:
Yes. I don't know if forests generate sinkholes, but if they do, this line is now redundant, and if they don't, then neither should spider pits.

Reply:
I'm not sure I follow that logic. If spider pits had sinkholes before, should they not still have them going forward? I haven't actually ventured into a spider pit area, but it seems intentional to have sinkhole traps for this area. Maybe someone more familiar with these can chime in though.

Reply:
I removed those sinkholes because there's a cavern under them. In reality, those sinkholes would've collapsed a large chunk of the cavern, destroying all the spider nests there (if my understanding about sinkholes is correct).

Reply:
I think this is out of scope here. I would re-add

```json
"traps": { ".": [ [ "tr_null", 20 ], "tr_sinkhole" ] },
```

There are a couple of issues with sinkholes. You are tackling forest generation here, which is awesome by the way, not sinkhole issues. I would focus on forest generation and leave the sinkholes for another PR.